src/util/pagination.tsx

```diff
@@ -17,18 +18,51 @@ export const paginationOptions = [
   },
 ];
 
-export const usePagination = (
-  data: MainTableRow[]
-): {
+export interface Pagination {
   pageData: MainTableRow[];
   currentPage: number;
   paginate: (pageNumber: number) => void;
   pageSize: number;
   setPageSize: (newSize: number) => void;
   totalPages: number;
-} => {
+  updateSort: (sort: string | null | undefined) => void;
+}
+
+export const usePagination = (
+  data: MainTableRow[],
+  defaultSort?: string,
+  defaultSortDirection?: SortDirection
+): Pagination => {
   const [_pageSize, _setPageSize] = useState(paginationOptions[0].value);
 
+  const [sort, setSort] = useState<string | null | undefined>(defaultSort);
+  const [sortDirection, setSortDirection] = useState(
+    defaultSortDirection ?? "ascending"
+  );
+
+  if (sort) {
+    data.sort((a, b) => {
+      const aVal = a.sortData ? (a.sortData[sort] as string) : "";
+      const bVal = b.sortData ? (b.sortData[sort] as string) : "";
+      if (aVal > bVal) {
+        return sortDirection === "ascending" ? 1 : -1;
+      }
+      if (aVal < bVal) {
+        return sortDirection === "ascending" ? -1 : 1;
+      }
+      return 0;
+    });
+  }
+
+  const updateSort = (newSort: string | null | undefined) => {
```

Comment:
See comment above, about the interface for this function:
```suggestion
  const updateSort = (newSort?: string) => {
```